BusyList/TaskRepository.cs

```diff
@@ -0,0 +1,72 @@
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace BusyList
+{
+    public class TaskRepository : ITaskRepository
+    {
+        private const string PATH = "database.json";
+
+        private readonly List<TaskItem> _items = new List<TaskItem>();
+
+        private int _idCounter = 1;
+
+        public TaskRepository()
+        {
+            if (File.Exists(PATH))
+            {
+                var text = File.ReadAllText(PATH);
+
+                _items = JsonConvert.DeserializeObject<List<TaskItem>>(text);
+
+                _idCounter = _items.Max(task => task.Id) + 1;
+            }
+        }
+        public TaskItem AddTask(AddTaskData data)
+        {
+            var item = new TaskItem(_idCounter++, data.Description);
+
+            _items.Add(item);
+
+            Save();
+
+            return item;
+        }
+
+        public void DeleteTask(TaskItem task)
+        {
+            Console.WriteLine($"Removed task #{task.Id}");
+
+            _items.Remove(task);
+
+            Save();
+        }
+
+        public IEnumerable<TaskItem> GetAll()
+        {
+            return _items;
+        }
+
+        public TaskItem GetTaskById(int id)
+        {
+            return _items.FirstOrDefault(x => x.Id == id);
+        }
+
+        public TaskItem UpdateTask(TaskItem task)
+        {
+
+            Save();
+
+            return task;
+        }
+        private void Save()
```

Comment:
we could add an append to file method insated of  cnverting entire task list everytime so we could call that in 
AddTask() ,  DeleteTask() , UpdateTask().
or we could call this method before application is closed

Reply:
That complicates the writing process massively. For now, lets keep it simple.